cohortextractor/main.py

```diff
@@ -13,24 +14,26 @@ def main(definition_path, output_file, backend_id, db_url):
     write_output(results, output_file)
 
 
-def load_cohort(definition):
-    cohort_definition = load_module_by_path(definition)
+def load_cohort(definition_path):
+    definition_module = load_module(definition_path)
     cohort_classes = [
         obj
-        for name, obj in inspect.getmembers(cohort_definition)
+        for name, obj in inspect.getmembers(definition_module)
         if inspect.isclass(obj)
     ]
     assert len(cohort_classes) == 1, "A study definition must contain one class only"
     return cohort_classes[0]
 
 
-def load_module_by_path(path):
-    # We have to jump through hoops here because the thing we've got in our hand is the full path of the file where the
-    # module is defined.
-    spec = importlib.util.spec_from_file_location("cohort_definition", path)
-    cohort_definition = importlib.util.module_from_spec(spec)
-    spec.loader.exec_module(cohort_definition)
-    return cohort_definition
+def load_module(definition_path):
+    definition_dir = definition_path.parent
+    module_name = definition_path.stem
+
+    # Add the directory containing the definition to the path so that the definition can import library modules from
+    # that directory
+    sys.path.append(str(definition_dir))
```

Comment:
Is it worth restoring the original `sys.path` value after the import? It's unlikely to cause us trouble, but it's always possible that a user might add a local module which shadows something we later want to import.
 
Genuine non-rhetorical question! The answer might be that it's not worth worrying about.

Reply:
Oh yes, nice!

Reply:
No. This doesn't work. Once something has been imported it's cached indefinitely. So removing the directory from the path doesn't have any effect (except for modules which are defined there but never loaded, which seems like a pretty extreme edge case) and prepending doesn't fix the problem that I had seen before.

I'm going to leave in the removal, because it just seems like the right thing to do from a semantic point of view, though.